src/core/score_map.cc

```diff
@@ -146,4 +146,31 @@ detail::SdsScorePair ScoreMap::iterator::BreakToPair(void* obj) {
   return detail::SdsScorePair(f, GetValue(f));
 }
 
+bool ScoreMap::iterator::ReallocIfNeeded(float ratio) {
+  // Unwrap all links to correctly call SetObject()
+  auto* ptr = curr_entry_;
+  while (ptr->IsLink())
```

Comment:
this code is incorrect. has it been tested somehow?

Reply:
We synced offline and also synced with vlad:

1. The `while (ptr->IsLink())` was added by vlad for "just in case". It's not really needed, we can just `ptr->AsLink()` directly if `ptr->IsLink()`
2. We do not iterate over all the links

We should also have some tooling around this. We should be able to collect info on how much each of the blocks if fragmented. That way, when we deploy a different algorithm for `fragmentation` would be able to compare it. (yes we do have the tests but they do not spit information for the whole data strucutre)